regipy/plugins/system/usbstor.py

```diff
@@ -0,0 +1,72 @@
+
+import logging
+
+from regipy.exceptions import RegistryKeyNotFoundException
+from regipy.hive_types import SYSTEM_HIVE_TYPE
+from regipy.plugins.plugin import Plugin
+from regipy.utils import convert_wintime
+
+logger = logging.getLogger(__name__)
+
+USBSTOR_KEY_PATH = r'Enum\USBSTOR'
+PROPERTIES_NAME_GUID = r'{540b947e-8b40-45bc-a8a2-6a0b894cbda2}'
+PROPERTIES_DATES_GUID = r'{83da6326-97a6-4088-9453-a1923f573b29}'
+
+
+class USBSTORPlugin(Plugin):
+    NAME = 'usbstor_plugin'
+    DESCRIPTION = "Parse the connected USB devices history"
+    COMPATIBLE_HIVE = SYSTEM_HIVE_TYPE
+
+    def run(self):
+        try:
+            for subkey_path in self.registry_hive.get_control_sets(USBSTOR_KEY_PATH):
+                usbstor_key = self.registry_hive.get_key(subkey_path)
+                for usbstor_drive in usbstor_key.iter_subkeys():
+                    try:
+                        disk, manufacturer, title, version = usbstor_drive.name.split('&')
+                    except ValueError:
+                        manufacturer = ""
+                        title = ""
+                        version = ""
+
+                    for serial_subkey in usbstor_drive.iter_subkeys():
+                        timestamp = convert_wintime(serial_subkey.header.last_modified, as_json=self.as_json)
+                        serial_number = serial_subkey.name
+
+                        device_guid_key = serial_subkey.get_subkey('Device Parameters').get_subkey('Partmgr')
```

Comment:
`get_subkey` might raise `NoRegistrySubkeysException`, I think it is better to wrap each get_subkey separately